File: RecyclerViewExample/app/src/main/java/edu/kiet/recyclerviewexample/RecyclerAdapter.java
package edu.kiet.recyclerviewexample;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.MyViewHolder> {
    Context context;
    int [] image;
    String [] imageName;
    LayoutInflater layoutInflater;
    public RecyclerAdapter( Context context, int[] image, String[] imageName) {
        this.context=context;
        this.image=image;
        this.imageName=imageName;
    }

    @NonNull
    @Override
    public RecyclerAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        layoutInflater= (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view =layoutInflater.inflate(R.layout.recyclerlayout,parent,false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerAdapter.MyViewHolder holder, int position) {
                    holder.image.setImageResource(image[position]);
                    holder.imageName.setText(imageName[position]);
                    holder.imageName.setOnClickListener(new View.OnClickListener() {
                        @Override
                        public void onClick(View v) {
                            Toast.makeText(context,"You clicked on "+imageName[position],Toast.LENGTH_SHORT).show();

                        }
                    });
    }

    @Override
    public int getItemCount() {
        return image.length;
    }
    class MyViewHolder extends RecyclerView.ViewHolder{
        ImageView image;
        TextView imageName;
        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            image=itemView.findViewById(R.id.imageView);
            imageName=itemView.findViewById(R.id.textView);
        }
    }
}

File: RecyclerViewWithKotlin/app/src/main/java/edu/kiet/recyclerviewwithkotlin/RecyclerAdapter.kt
package edu.kiet.recyclerviewwithkotlin

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView

class RecyclerAdapter(context:Context,imageName: ArrayList<Int>,imageTitle: ArrayList<String>):RecyclerView.Adapter<RecyclerAdapter.MyViewHolder>() {
    var context:Context=context;
    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
            var iname: ImageView =itemView.findViewById(R.id.imageName);
        var ititle: TextView =itemView.findViewById(R.id.imageTitle);
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        var view:View =LayoutInflater.from(parent.context).inflate(R.layout.recyclerlayout,parent,false);
        return MyViewHolder(view);
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.iname.setImageResource(imageName[position]);
        holder.ititle.setText(imageTitle[position]);
        holder.iname.setOnClickListener(View.OnClickListener { v: View? ->
            Toast.makeText(context,"You checked on "+imageTitle[position], Toast.LENGTH_SHORT).show();
        })
    }

    override fun getItemCount(): Int {
        return imageName.size;
    }

}
